dataland-frontend/src/components/forms/parts/kpiSelection/KPIfieldSet.vue

```diff
@@ -0,0 +1,166 @@
+              :options="['None...', ...reportsName]"
+            />
+          </div>
+          <div>
+            <UploadFormHeader :name="kpiNameMappings.page ?? ''" :explanation="kpiInfoMappings.page ?? ''" />
+            <FormKit
+              outer-class="w-100"
+              :disabled="!dataPointIsAvailable"
+              v-model="currentPageValue"
+              type="number"
+              name="page"
+              placeholder="Page"
+              validation-label="Page"
+            />
+          </div>
+        </div>
+      </FormKit>
+    </div>
+
+    <!-- Data quality -->
+    <div class="form-field">
+      <UploadFormHeader name="Data quality" explanation="Data quality" />
+      <div class="md:col-6 col-12 p-0">
+        <FormKit
+          :disabled="!dataPointIsAvailable"
+          type="select"
+          v-model="currentQualityValue"
+          name="quality"
+          :validation="dataPointIsAvailable ? 'required' : ''"
+          validation-label="Data quality"
+          placeholder="Data quality"
+          :options="dataQualityList"
+        />
+      </div>
+    </div>
+  </div>
+  <div class="form-field">
+    <FormKit
+      type="textarea"
+      name="comment"
+      placeholder="(Optional) Add comment that might help Quality Assurance to approve the datapoint. "
+    />
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from "vue";
+import InputSwitch from "primevue/inputswitch";
+import UploadFormHeader from "@/components/forms/parts/UploadFormHeader.vue";
+import { FormKit } from "@formkit/vue";
+
+export default defineComponent({
+  name: "KPIfieldSet",
+  components: { UploadFormHeader, FormKit, InputSwitch },
+  emits: ["dataPointAvailableToggle"],
+  data: () => ({
+    dataPointIsAvailable: true,
+    dataQualityList: ["NA", "Audited", "Reported", "Estimated", "Incomplete"],
+    currentMainValue: "",
+    currentReportValue: "",
+    currentQualityValue: "",
+    currentPageValue: "",
+    qualityValueBeforeDataPointWasDisabled: "",
+    qualityValueWhenDataPointIsDisabled: "",
+    pageValueWhenDataPointIsDisabled: "",
+    reportValueWhenDataPointIsDisabled: "",
+    mainValueWhenDataPointIsDisabled: "",
+  }),
+  watch: {
+    dataPointIsAvailable(newValue: boolean) {
+      if (!newValue) {
+        this.qualityValueBeforeDataPointWasDisabled = this.currentQualityValue;
+        this.pageValueWhenDataPointIsDisabled = this.currentPageValue;
+        this.reportValueWhenDataPointIsDisabled = this.currentReportValue;
+        this.mainValueWhenDataPointIsDisabled = this.currentMainValue;
+        this.currentQualityValue = "NA";
+      } else {
+        this.currentQualityValue = this.qualityValueBeforeDataPointWasDisabled;
+        this.currentPageValue = this.pageValueWhenDataPointIsDisabled;
+        this.currentReportValue = this.reportValueWhenDataPointIsDisabled;
+        this.currentMainValue = this.mainValueWhenDataPointIsDisabled;
+      }
```

Comment:
Have you added this watcher because of some cornercase that I have overlooked?
Because the component works also without this watcher for me.
This is expected because even though those fields are not shown/disabled anymore if you click the "Datapoint available" toggle, their values are still stored inside of the `data` object.
That's why I expect to be there when I reactivate the toggle. And this is also the case when I tried it.

Reply:
Their values are still stored inside of the data object, and they are sent even if they are invisible

Reply:
I think I still don't get it

Reply:
Marcin explained this to me and the problem behind it.

=> TODO:    We need to do some research if we can solve that problem some way different. 
If we don't manage to solve it in a different way, we will need watchers like this for every "disable-able" field we will have in any form in the future!

Reply:
passed to @stewal19

Reply:
remove unnecessary watchers